app/controllers/permits_controller.rb

```diff
@@ -44,6 +44,11 @@ def load_adapter
       flash.alert = 'Adapter not found or not available'
       redirect_to(permits_path) && return
     end
+    user_country = current_user.organisation.country_id
+    unless organisation.adapter.countries_with_access_ids.include?(user_country)
+      flash.alert = 'You are not allow to acces this Adapter'
```

Comment:
Make the error messages more human readable, nobody is going to have a clue what an adapter is :) sth simpler like "Web Service not defined" / "Web Service access denied"